prefixfilter.go

```diff
@@ -0,0 +1,48 @@
+package complete
+
+import (
+	"strings"
+)
+
+//PrefixFilter filters a predictor's options based on the prefix
+type PrefixFilter interface {
+	FilterPrefix(str, prefix string) bool
+}
+
+//PrefixFilteringPredictor is a Predictor that also implements PrefixFilter
+type PrefixFilteringPredictor struct {
```

Comment:
Maybe this struct definition is not necessary?
If anyone wants to use the new interface, They will just make sure that the predictor also implements the new interface?

Reply:
I think this will be useful in instances where somebody wants to use an existing Predictor with a non-default PrefixFilter.

An example would be if I want to use a SetPredictor with a case insensitive matcher I would do something like:

```golang
&complete.PrefixFilteringPredictor{
	PrefixFilterFunc: complete.CaseInsensitivePrefixFilter,
	Predictor:        complete.PredictSet("foo", "bar", "baz", )
}
```

Without it, I end up implementing PrefixFilteringPredictor in my own codebase. Admittedly it's pretty simple to implement though.

Reply:
Following https://github.com/posener/complete/pull/100/files#r301385506, this won't be needed.